app/celery/contact_information_tasks.py

```diff
@@ -24,55 +38,197 @@
+        Exception: If an unhandled exception occurs.
+
+    Returns:
+        None
+    """
     current_app.logger.info('Looking up contact information for notification_id: %s.', notification_id)
 
     notification = get_notification_by_id(notification_id)
     recipient_identifier = notification.recipient_identifiers[IdentifierType.VA_PROFILE_ID.value]
+    should_send = notification.default_send
 
     try:
-        if EMAIL_TYPE == notification.notification_type:
-            recipient = va_profile_client.get_email(recipient_identifier)
-        elif SMS_TYPE == notification.notification_type:
-            recipient = va_profile_client.get_telephone(recipient_identifier)
+        if is_feature_enabled(FeatureFlag.VA_PROFILE_V3_COMBINE_CONTACT_INFO_AND_PERMISSIONS_LOOKUP):
+            result = get_profile_result(notification, recipient_identifier)
+            recipient = result.recipient
+            should_send = result.communication_allowed
+            permission_message = result.permission_message
         else:
-            raise NotImplementedError(
-                f'The task lookup_contact_info failed for notification {notification_id}. '
-                f'{notification.notification_type} is not supported'
+            recipient = get_recipient(
+                notification.notification_type,
+                notification_id,
+                recipient_identifier,
             )
-    except (Timeout, VAProfileRetryableException) as e:
-        if can_retry(self.request.retries, self.max_retries, notification_id):
-            current_app.logger.warning('Unable to get contact info for notification id: %s, retrying', notification_id)
+    except Exception as e:
+        handle_lookup_contact_info_exception(self, notification, recipient_identifier, e)
+
+    notification.to = recipient
+    dao_update_notification(notification)
+
+    if not should_send:
+        handle_communication_not_allowed(notification, recipient_identifier, permission_message)
+
+
+def get_recipient(
+    notification_type: str,
+    notification_id: str,
+    recipient_identifier: RecipientIdentifier,
+) -> str:
+    """
+    Retrieve the recipient email or phone number.
+
+    Args:
+        notification_type (str): The type of recipient info requested.
+        notification_id (str): The notification ID associated with this request.
+        recipient_identifier (RecipientIdentifier): The VA profile ID to retrieve the profile for.
+
+    Returns:
+        str: The recipient email or phone number.
+    """
+    if notification_type == EMAIL_TYPE:
+        return va_profile_client.get_email(recipient_identifier)
+    elif notification_type == SMS_TYPE:
+        return va_profile_client.get_telephone(recipient_identifier)
+    else:
+        raise NotImplementedError(
+            f'The task lookup_contact_info failed for notification {notification_id}. '
+            f'{notification_type} is not supported'
+        )
+
+
+def get_profile_result(
```

Comment:
This is the new functionality that replaces `get_recipient`.  It calls the new functions, `get_email_with_permission` and `get_telephone_with_permission` instead of `get_email` and `get_telephone`.